hydromt/components/vector.py

```diff
@@ -175,8 +188,8 @@ def set(
     def read(
         self,
         *,
-        filename: Optional[str] = None,
-        geometry_filename: Optional[str] = None,
+        filename: Optional[str] = "vector/vector.nc",
+        geometry_filename: Optional[str] = "vector/vector.geojson",
```

Comment:
This is old style again. We have come to the decision that the optionals are by default None and will then use the self.filename, which is by default DEFAULT_FILENAME

Reply:
It's back to old style because for these components, we should allow for None in one of the files (ie output file can be netcdf only or geometry only or netcdf+geometry). With the default filenames, it is too tough to specify that you want None for one of them so I went back to the old behavior to be sure a user can overwrite defaults easily if he is not happy with them without having to change the component instantiation in the yaml which I find complex.

Reply:
Alright, that makes sense. In that case I think we should continue with this in all other components as well and make some of them non-optional. @savente93 , do you agree?